llvm/lib/Transforms/Instrumentation/InstrProfiling.cpp

```diff
@@ -937,6 +949,44 @@ Value *InstrLowerer::getCounterAddress(InstrProfCntrInstBase *I) {
   return Builder.CreateIntToPtr(Add, Addr->getType());
 }
 
+Function *InstrLowerer::createRMWOrFunc() {
+  auto &Ctx = M.getContext();
+  auto *Int8Ty = Type::getInt8Ty(Ctx);
+  // void alwaysinline rmw_or(ptr, i8)
+  Function *Fn = Function::Create(
+      FunctionType::get(Type::getVoidTy(Ctx),
+                        {PointerType::getUnqual(Ctx), Int8Ty}, false),
+      Function::LinkageTypes::PrivateLinkage, "rmw_or", M);
```

Comment:
Is there a convention for naming compiler-generated functions? Will there be problems if a user creates a function with the same name? Or is this ok becase the linkage is private?

Reply:
I suppose private named functions wouldn't clash. Let me reconfirm later.
(I guess it might be unnamed)

Reply:
I've reconfirmed this won't clash. If I called `createRMWOrFunc()` twice, I got both `@rmw_or` and `@rmw_or.1`.

This may work with unnamed.